### ISSUE TITLE
adapt model to the trainer API

### ISSUE THREAD

**Issue by not-lain** [jasonppy/VoiceCraft#125]
not an issue but rather an improvement.
the trainer API offers a set of easy-to-implement functionalities plus it's compatible with the multi-GPU training by default.
I described the process of how to make it work here: https://huggingface.co/blog/not-lain/trainer-api-and-mixin-classes#trainer-api-compatibility.
in theory, we should add the labels parameter to the forward method and return a dictionary with the loss in it if the labels parameter is present, a ruff estimation of the final function would look something like this :
```python
    def forward(self, batch,labels=None):
        # stuff here
        if labels != None :
          loss = # logic to calculate the loss here
          return {"loss": loss}
        return logits
```

this is an optional feature, feel free to close it if it's out of scope.
